### ISSUE TITLE
MethodInfo.ReturnParameter.IsDefined(type) throws IndexOutOfRangeException

### ISSUE THREAD

**Issue by StevenBonePgh** [xamarin/xamarin-android#2175]
### Steps to Reproduce

1. MethodInfo.ReturnParameter.IsDefined(type) throws IndexOutOfRangeException
```
Type type = typeof(object);
MethodInfo method = type.GetMethod("ToString");
ParameterInfo paramInfo = method.ReturnParameter;
bool defined = paramInfo.IsDefined(typeof(Attribute)); //throws
```
### Expected Behavior
For the sample above, `defined` should be false.
### Actual Behavior
Exception thrown:
```
{System.IndexOutOfRangeException} "Index was outside the bounds of the array."
  at System.Attribute.InternalParamIsDefined (System.Reflection.ParameterInfo parameter, System.Type attributeType, System.Boolean inherit) [0x0003b] in <43dbbdc147f2482093d8409abb04c233>:0 
  at System.Attribute.IsDefined (System.Reflection.ParameterInfo element, System.Type attributeType, System.Boolean inherit) [0x00071] in <43dbbdc147f2482093d8409abb04c233>:0 
  at System.Attribute.IsDefined (System.Reflection.ParameterInfo element, System.Type attributeType) [0x00000] in <43dbbdc147f2482093d8409abb04c233>:0 
  at System.Reflection.CustomAttributeExtensions.IsDefined (System.Reflection.ParameterInfo element, System.Type attributeType) [0x00000] in <43dbbdc147f2482093d8409abb04c233>:0 
```
First seen in issue [Code-Sharp/WampSharp/issues/268](https://github.com/Code-Sharp/WampSharp/issues/268). Workaround was to replace usage with `MethodInfo.ReturnParameter.GetCustomAttribute<>()` This is likely an issue with the underlying implementation on Mono, although I have only reported it here as I can reproduce it easily on current toolchain for Xamarin.Android and do not have a suitable Mono playground.

### Version Information

Microsoft Visual Studio Enterprise 2017 
Version 15.8.3
VisualStudio.15.Release/15.8.3+28010.2019
Microsoft .NET Framework
Version 4.7.03056

Installed Version: Enterprise

Architecture Diagrams and Analysis Tools   00369-60000-00001-AA256
Microsoft Architecture Diagrams and Analysis Tools

Visual C++ 2017   00369-60000-00001-AA256
Microsoft Visual C++ 2017

ASP.NET and Web Tools 2017   15.8.05077.0
ASP.NET and Web Tools 2017

ASP.NET Core Razor Language Services   15.8.31590
Provides languages services for ASP.NET Core Razor.

ASP.NET Web Frameworks and Tools 2017   5.2.60618.0
For additional information, visit https://www.asp.net/

Atomineer Pro Documentation   1.0
Source-code documentation-comment creation and maintenance, and other documentation-related tools.

Azure App Service Tools v3.0.0   15.8.05023.0
Azure App Service Tools v3.0.0

C# Tools   2.9.0-beta8-63208-01
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.1.49 (remotes/origin/d15-8@ee674f3)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

GitHub.VisualStudio   2.5.5.3913
A Visual Studio Extension that brings the GitHub Flow into Visual Studio.

IntelliCode Extension   1.0
IntelliCode Extension

IntellicodePackage Extension   1.0
IntellicodePackage Visual Studio Extension Detailed Info

JavaScript Language Service   2.0
JavaScript Language Service

JavaScript Project System   2.0
JavaScript Project System

JetBrains ReSharper Ultimate 2018.2.1   Build 182.0.20180829.63240
JetBrains ReSharper Ultimate package for Microsoft Visual Studio. For more information about ReSharper Ultimate, visit http://www.jetbrains.com/resharper. Copyright © 2018 JetBrains, Inc.

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of continuous build integration and continuous build delivery from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   1.0
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

MLGen Package Extension   1.0
MLGen Package Visual Studio Extension Detailed Info

Mono Debugging for Visual Studio   4.11.8-pre (009db7d)
Support for debugging Mono processes with Visual Studio.

Node.js Tools   1.4.20802.1 Commit Hash:97e1085d8b4b8e3e51c398e910177f87e86d135e
Adds support for developing and debugging Node.js apps in Visual Studio

NuGet Package Manager   4.6.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

OzCode   1.0
Your road to Magical Debugging. http://www.oz-code.com

OzCodePackage Extension   1.0
OzCodePackage Visual Studio Extension Detailed Info

Pendletron.Vsix.LocateInTFS   1.0
Opens the source control explorer for the currently selected item

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

ResourcePackage Extension   1.0
ResourcePackage Visual Studio Extension Detailed Info

ResourcePackage Extension   1.0
ResourcePackage Visual Studio Extension Detailed Info

Sandcastle Help File Builder   SHFB
Visual Studio integration for the Sandcastle Help File Builder.
https://GitHub.com/EWSoftware/SHFB

SQL Server Data Tools   15.1.61808.07020
Microsoft SQL Server Data Tools

SQL Server Reporting Services   14.0.1016.285
Microsoft SQL Server Reporting Services Designers 
Version 14.0.1016.285

SQLite & SQL Server Compact Toolbox   4.7
SQLite & SQL Server Compact Toolbox adds scripting, import, export, rename, query execution and much more to SQL Server Compact & SQLite Data Connections.

tangible T4 Editor   2.4.0
tangible engineering GmbH

TFS Source Control Explorer Extension   1.0
Visual Studio Extension for Team Foundation Server Source Control Explorer

TypeScript Tools   15.8.20801.2001
TypeScript Tools for Microsoft Visual Studio

Visual Assist   
For more information about Visual Assist, see the Whole Tomato Software website at http://www.WholeTomato.com. Copyright (c) 1997-2018 Whole Tomato Software, Inc.

Visual Basic Tools   2.9.0-beta8-63208-01
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual C++ for Cross Platform Mobile Development (Android)   15.0.27924.00
Visual C++ for Cross Platform Mobile Development (Android)

Visual C++ for Cross Platform Mobile Development (iOS)   15.0.27825.00
Visual C++ for Cross Platform Mobile Development (iOS)

Visual F# Tools 10.2 for F# 4.5   15.8.0.0.  Commit Hash: c55dd2c3d618eb93a8d16e503947342b1fa93556.
Microsoft Visual F# Tools 10.2 for F# 4.5

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Tools for Apache Cordova   15.123.7408.1
Visual Studio Tools for Apache Cordova

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

Visual Studio Tools for Universal Windows Apps   15.0.28010.00
The Visual Studio Tools for Universal Windows apps allow you to build a single universal app experience that can reach every device running Windows 10: phone, tablet, PC, and more. It includes the Microsoft Windows 10 Software Development Kit.

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

WiX Toolset Visual Studio Extension   0.9.21.62588
WiX Toolset Visual Studio Extension version 0.9.21.62588
Copyright (c) .NET Foundation and contributors. All rights reserved.

Xamarin   4.11.0.758 (d15-8-xcode10@309af5cf6)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   4.14.253 (f2941f7cb)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   1.1.116 (9619170)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   9.0.0.19 (HEAD/a8a3b0ec7)
Xamarin.Android Reference Assemblies and MSBuild support.

Xamarin.iOS and Xamarin.Mac SDK   11.99.3.539 (1d5c334)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

**Comment by StevenBonePgh** [comment]
This is being resolved in mono/mono#12069. This issue should be closed once it makes it to RTM in Xamarin.

**Comment by jpobst** [comment]
We currently ship Mono from 2020, so I think this has been fixed.
